cms/djangoapps/contentstore/views/course.py

```diff
@@ -575,6 +582,40 @@ def format_library_for_view(library):
         u'optimization_enabled': optimization_enabled
     })
 
+@login_required
+@ensure_csrf_cookie
+def library_listing(request):
+    """
+    List all Libraries available to the logged in user
+    """
+    libraries = _accessible_libraries_iter(request.user) if LIBRARIES_ENABLED else []
+
+    def format_library_for_view(library):
```

Comment:
we are duplicating a lot of code here. An alternative approch should be reuse `format_in_process_course_view` and  `format_library_for_view` already defined inside `course_listing`. 

We would need to extract them out of `course_listing` because now this is not the only method needing them.

Once you do that use them in both views.